src/resources/views/crud/fields/radio.blade.php

```diff
@@ -2,6 +2,12 @@
 @php
     $optionValue = old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? '';
 
+
+    //check if attribute is casted, if it is, we get back un-casted values
+    if(in_array($field['name'], array_keys($crud->model->getCasts()))) {
```

Comment:
```suggestion
    if(Arr::get($crud->model->getCasts(), $field['name']) === 'boolean') {
```

It's better to check if the cast is to `boolean`, otherwise if a developer set the cast to anything else, like `integer` (very unlikely, but 🤷‍♂️), when setting `$optionValue = 1`, the validation `$optionValue === true` is going to be `false`, therefore `0` instead of `1`.

Reply:
Bump @pxpm . That sounds like a good idea, right?